Add bios disassembly file generation rules.

nrndda opened the pull request:
Added bios disassembly file generation. Very useful for debugging.

nrndda committed:
Add bios disassembly file generation rules.

enjoy-digital commented:
Thanks, this is useful for debugging BIOS/early boot issues.

I updated the PR a bit with a follow-up commit: 1edc341f5.

The disassembly generation is now opt-in instead of being part of the default BIOS build. This avoids generating an extra file on every normal build/CI run, while still making it easy to use when needed.

After building the BIOS, the user can generate the disassembly from the BIOS build directory with:

  ```sh
  make dis
  ```

  or:

  make bios.dis

This creates bios.dis from bios.elf using the target objdump.

enjoy-digital committed:
software: Make BIOS disassembly opt-in

Keep bios.dis generation available for debugging, but do not include it in the default BIOS build target.

The default all target remains focused on bios.bin and memory usage reporting, while users who need the disassembly can explicitly run make bios.dis or make dis from the BIOS build directory.
